packaging/release.py

```diff
@@ -0,0 +1,1410 @@
+
+
+def get_next_release_date(start: datetime.date, step: int, after: datetime.date) -> datetime.date:
+    """Return the next release date."""
+    if start > after:
+        raise ValueError(f"{start=} is greater than {after=}")
+
+    current_delta = after - start
+    release_delta = datetime.timedelta(days=(math.floor(current_delta.days / step) + 1) * step)
+
+    release = start + release_delta
+
+    return release
+
+
+def create_template_environment() -> jinja2.Environment:
+    """Create and return a jinja2 environment."""
+    env = jinja2.Environment()
+    env.filters.update(
+        basename=os.path.basename,
+    )
+
+    return env
+
+
+def create_github_release_notes(upstream: Remote, repository: str, version: Version, validate: bool) -> str:
+    """Create and return GitHub release notes."""
+    env = create_template_environment()
+    template = env.from_string(GITHUB_RELEASE_NOTES_TEMPLATE)
+
+    variables = dict(
+        version=version,
+        releases=get_release_artifact_details(repository, version, validate),
+        changelog=f"https://github.com/{upstream.user}/{upstream.repo}/blob/v{ version }/changelogs/CHANGELOG-v{ version.major }.{ version.minor }.rst",
+    )
+
+    release_notes = template.render(**variables).strip()
+
+    return release_notes
+
+
+def create_release_announcement(upstream: Remote, repository: str, version: Version, validate: bool) -> ReleaseAnnouncement:
+    """Create and return a release announcement message."""
+    env = create_template_environment()
+    subject_template = env.from_string(RELEASE_ANNOUNCEMENT_SUBJECT_TEMPLATE)
+    body_template = env.from_string(RELEASE_ANNOUNCEMENT_BODY_TEMPLATE)
+
+    today = datetime.datetime.now(tz=datetime.timezone.utc).date()
+
+    variables = dict(
+        version=version,
+        info=dict(
+            name="ansible-core",
+            short=f"{version.major}.{version.minor}",
+            releases=get_release_artifact_details(repository, version, validate),
+        ),
+        next_rc=get_next_release_date(datetime.date(2021, 8, 9), 28, today),
+        next_ga=get_next_release_date(datetime.date(2021, 8, 16), 28, today),
+        rc=version.pre and version.pre[0] == "rc",
+        beta=version.pre and version.pre[0] == "b",
+        alpha=version.pre and version.pre[0] == "a",
+        major=version.micro == 0,
+        upstream=upstream,
+    )
+
+    if version.pre and version.pre[0] in ("a", "b"):
+        display.warning("The release announcement template does not populate the date for the next release.")
+
+    subject = subject_template.render(**variables).strip()
+    body = body_template.render(**variables).strip()
+
+    message = ReleaseAnnouncement(
+        subject=subject,
+        body=body,
+    )
+
+    return message
+
+
+# endregion
+# region Templates
+
+
+RELEASE_ANNOUNCEMENT_RECIPIENTS = (
+    "ansible-announce@googlegroups.com",
+    "ansible-project@googlegroups.com",
+    "ansible-devel@googlegroups.com",
+)
```

Comment:
Since we're using mailto, this is of course editable before sending, but just as an FYI, for pre-releases we've only been sending to ansible-devel.  Only GAs have been going to all 3.

Reply:
I've split the recipient lists into two, one for final releases and one for development releases.